File: 6. Python/EX56.py
# Solicita a quantidade de linhas da matriz
m = int(input("# Qual a quantidade de linhas da matriz: "))

# Solicita a quantidade de colunas da matriz
n = int(input("# Qual a quantidade de colunas da matriz: "))

idAux = 0                                       # Inicializa índice para negativos
aux = [0 for i in range(m * n)]                 # Cria lista para armazenar negativos
mat = [[0 for i in range(n)] for i in range(m)] # Inicializa a matriz

# Preenche a matriz com entradas do usuário
for i in range(m):

    for j in range(n):
        mat[i][j] = int(input(f"# Elemento [{i},{j}]: "))  # Captura elemento

        if mat[i][j] < 0:           # Checa se o elemento é negativo
            aux[idAux] = mat[i][j]  # Armazena valor negativo
            idAux += 1              # Incrementa o índice

# Exibe os valores negativos armazenados
print("\n# VALORES NEGATIVOS: ")
for i in range(idAux):
    print(f"# {aux[i]}")  # Imprime cada negativo

File: 4. Java/EX56.java
package Apostila;

import java.util.Scanner;

public class EX56
{
    public static void main(String[] args)
    {
        int idAux = 0, m, n; // Variáveis auxiliares
        
        // Entrada de dados
        try (Scanner ent = new Scanner(System.in)) 
        {
            // Solicita linhas e colunas
            System.out.print("# Qual a quantidade de linhas da matriz: ");
            m = ent.nextInt();
            
            System.out.print("# Qual a quantidade de colunas da matriz: ");
            n = ent.nextInt();
            
            int [][] mat = new int [m][n];  // Matriz
            int [] aux = new int [m];       // Valores negativos
            
            // Preenche matriz e guarda negativos
            for (int i = 0; i < m; i++)
            {
                for (int j = 0; j < n; j++)
                {
                    System.out.printf("# Elemento [%d, %d]: ", i, j);
                    mat[i][j] = ent.nextInt();
                    
                    if (mat[i][j] < 0)
                    {
                        aux[idAux] = mat[i][j];
                        idAux++;
                    }
                }
            }
            
            // Exibe negativos
            System.out.println("\n# VALORES NEGATIVOS:");
            for (int i = 0; i < idAux; i++)
            {
                System.out.println("# " + aux[i]);
            }
        }
    }
}
